src/S3/ObjectCopier.php

```diff
@@ -75,10 +77,12 @@ public function __construct(
      * Perform the configured copy asynchronously. Returns a promise that is
      * fulfilled with the result of the CompleteMultipartUpload or CopyObject
      * operation or rejected with an exception.
+     *
+     * @return Coroutine|PromiseInterface
      */
     public function promise()
     {
-        return \GuzzleHttp\Promise\Coroutine::of(function () {
```

Comment:
Is there a reason you removed the first bit of the namespace from the PHPDoc?  I'm not a PHPDoc expert, so maybe I'm missing something, but I'd think that the way they're displayed to a developer varies by IDE.  I'd think it would be best to keep as much info as possible.  

Is it a safe assumption the full namespace always get picked up from the use statement above?

Reply:
I rebased the PR and did it now in 3 separate commits to better explain why. 

The reason why I cleaned this up is because the FQCN looks bad and are probably added by mistake. 

The rest of the code base imports everything. Let's also do it here.

And yes, it's safe to put short class names in PHPDoc when they are imported at the top.